compactor/src/parquet_file_lookup.rs

```diff
@@ -88,17 +88,20 @@ impl ParquetFilesForCompaction {
         let mut level_2 = Vec::with_capacity(parquet_files.len());
 
         for parquet_file in parquet_files {
+            // Estimate the bytes DataFusion needs when scan this file
             let estimated_arrow_bytes = partition.estimated_arrow_bytes(
                 min_num_rows_allocated_per_record_batch_to_datafusion_plan,
                 parquet_file.row_count,
             );
+            // Add bytes need to read this file into memory
+            let total_estimated_bytes = estimated_arrow_bytes + parquet_file.file_size_bytes as u64;
```

Comment:
Since we are trying to be conservative, I suggest we consider estimating the RAM required to store the output parquet file as well. 

Ideally we'll make the plans fully streaming at some point, but for now they need to buffer the input parquet files in RAM as well as buffer the output files in RAM.

Thus, perhaps we might want to change this estimate to 

```suggestion
            // as well as the bytes needed to buffer the output
            let total_estimated_bytes = estimated_arrow_bytes + (2 * parquet_file.file_size_bytes as u64);
```